src/metalperformanceshaders.cs

```diff
@@ -4203,159 +6066,2016 @@ interface MPSImageReduceUnary {
+	}
+
+	[TV (12,0), Mac (10,14, onlyOn64: true), iOS (12,0)]
+	[BaseType (typeof (MPSCnnKernel), Name = "MPSCNNYOLOLoss")]
+	[DisableDefaultCtor]
+	interface MPSCnnYoloLoss {
+
+		[Export ("lossXY", ArgumentSemantic.Retain)]
+		MPSCnnLoss LossXY { get; }
+
+		[Export ("lossWH", ArgumentSemantic.Retain)]
+		MPSCnnLoss LossWH { get; }
+
+		[Export ("lossConfidence", ArgumentSemantic.Retain)]
+		MPSCnnLoss LossConfidence { get; }
+
+		[Export ("lossClasses", ArgumentSemantic.Retain)]
+		MPSCnnLoss LossClasses { get; }
+
+		[Export ("scaleXY")]
+		float ScaleXY { get; }
+
+		[Export ("scaleWH")]
+		float ScaleWH { get; }
+
+		[Export ("scaleNoObject")]
+		float ScaleNoObject { get; }
+
+		[Export ("scaleObject")]
+		float ScaleObject { get; }
+
+		[Export ("scaleClass")]
+		float ScaleClass { get; }
+
+		[Export ("minIOUForObjectPresence")]
+		float MinIouForObjectPresence { get; }
+
+		[Export ("maxIOUForObjectAbsence")]
+		float MaxIouForObjectAbsence { get; }
+
+		[Export ("reductionType")]
+		MPSCnnReductionType ReductionType { get; }
+
+		[Export ("numberOfAnchorBoxes")]
+		nuint NumberOfAnchorBoxes { get; }
+
+		[Export ("anchorBoxes", ArgumentSemantic.Retain)]
+		NSData AnchorBoxes { get; }
+
+		[Export ("initWithDevice:lossDescriptor:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, MPSCnnYoloLossDescriptor lossDescriptor);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder decoder, IMTLDevice device);
+
+		[Export ("encodeToCommandBuffer:sourceImage:labels:destinationImage:")]
+		void Encode (IMTLCommandBuffer commandBuffer, MPSImage sourceImage, MPSCnnLossLabels labels, MPSImage destinationImage);
+
+		[Export ("encodeToCommandBuffer:sourceImage:labels:")]
+		MPSImage Encode (IMTLCommandBuffer commandBuffer, MPSImage sourceImage, MPSCnnLossLabels labels);
+
+		[Export ("encodeBatchToCommandBuffer:sourceImages:labels:destinationImages:")]
+		void EncodeBatch (IMTLCommandBuffer commandBuffer, NSArray<MPSImage> sourceImage, NSArray<MPSCnnLossLabels> labels, NSArray<MPSImage> destinationImage);
+
+		[Export ("encodeBatchToCommandBuffer:sourceImages:labels:")]
+		NSArray<MPSImage> EncodeBatch (IMTLCommandBuffer commandBuffer, NSArray<MPSImage> sourceImage, NSArray<MPSCnnLossLabels> labels);
+	}
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[BaseType (typeof (MPSNNBinaryGradientState), Name = "MPSCNNArithmeticGradientState")]
+	[DisableDefaultCtor]
+	interface MPSCnnArithmeticGradientState {
+
+	}
+
+	[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+	[BaseType (typeof (MPSCnnBinaryKernel), Name = "MPSCNNArithmetic")]
+	[DisableDefaultCtor]
+	interface MPSCnnArithmetic {
+
+		// Inlined but you are not supposed to use this class, only its subclasses
```

Comment:
You could make the constructors protected. That way only subclasses would be able to use them.

Reply:
I would normally agree but subclasses already inline the constructor that is available for them and it is not always both of these.